libcxx/test/libcxx/algorithms/alg.sorting/assert.sort.invalid_comparator/assert.sort.invalid_comparator.oob.pass.cpp

```diff
@@ -0,0 +1,72 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+// REQUIRES: has-unix-headers
+// UNSUPPORTED: c++03, c++11, c++14, c++17
+// In the debug mode, the comparator validations will notice that it doesn't satisfy strict weak ordering before the
+// algorithm actually runs and goes out of bounds, so the test will terminate before the tested assertions are
+// triggered.
+// UNSUPPORTED: libcpp-hardening-mode=none, libcpp-hardening-mode=debug
+// XFAIL: libcpp-hardening-mode=debug && availability-verbose_abort-missing
```

Comment:
This XFAIL is superfluous since the test is `UNSUPPORTED` in the debug mode.

Reply:
I deliberately wrote it that way with the idea that it's easier to just blindly annotate all hardening tests; it's also a little more future-proof if a check gets reclassified. I plan to add a Lit feature to cover all the `check_assertion.h`-related feature together, which will hopefully make this moot. Unless you feel strongly about it, I'd rather leave this check as is.